src/exchange.ts

```diff
@@ -0,0 +1,101 @@
+import BN from "bn.js";
+import { TokenInfo } from "../src";
+import { BatchExchangeInstance } from "../types/truffle-typings";
+
+const MAXU32 = new BN(2).pow(new BN(32)).sub(new BN(1));
+
+export async function fetchTokenInfoFromExchange(
+  exchange: BatchExchangeInstance,
+  tokenIds: number[],
+  artifacts: Truffle.Artifacts,
+  fallbackSymbolName = "UNKNOWN",
+  fallbackDecimals = -1,
+): Promise<Map<number, TokenInfo>> {
+  const ERC20 = artifacts.require("ERC20Detailed");
+  // Fetching token data from EVM
+  const tokenObjects: Map<number, TokenInfo> = new Map();
+  for (const id of tokenIds) {
+    const tokenAddress = await exchange.tokenIdToAddressMap(id);
+    let tokenInfo;
+    try {
+      const tokenInstance = await ERC20.at(tokenAddress);
+      tokenInfo = {
+        id: id,
+        symbol: await tokenInstance.symbol.call(),
+        decimals: (await tokenInstance.decimals.call()).toNumber(),
+        address: tokenAddress,
+      };
+    } catch (err) {
+      tokenInfo = {
+        id: id,
+        symbol: fallbackSymbolName,
+        decimals: fallbackDecimals,
+        address: tokenAddress,
+      };
+    }
```

Comment:
@nlordell - I have two questions here. 

1. Do you happen to know why I do not error with `.call` here (say in `await tokenInstance.symbol.call()`) but had to remove all occurrences of this in the truffle test files?

2. Could I avoid this try-catch block by simply doing something like

```ts
symbol: await tokenInstance.symbol.call() || fallbackSymbolName
```

Or maybe something with a ?

Reply:
Also, doesn't this fallback functionality seem 🐟 ? Like what do we really plan to do when the decimals are wrong? Just make up some number and "go with it"? Seems silly to me. Also notice that if the token symbol doesn't exist, we just opt for returning the fallback decimals. 

I do not like this try-catch one little bit.

Reply:
I guess this is just a generic way of returning a Void/Null Token as a sort of "Token Not Found Error"

Reply:
Maybe `await tokenInstance.symbol?.call() || fallbackSymbolName`? And I think ERC20 doesn't require `decimals` right?